ui/src/components/WMWorkerPoolsTable/index.jsx

```diff
@@ -263,47 +257,34 @@ export default class WorkerManagerWorkerPoolsTable extends Component {
   };
 
   render() {
-    const { workerPools, searchTerm, includeDeleted } = this.props;
+    const { onPageChange, workerPoolsConnection } = this.props;
     const {
       sortBy,
       sortDirection,
       dialogState: { open, error, title, confirmText, body },
     } = this.state;
-    const sortedWorkerPools = this.sortWorkerPools(
-      workerPools,
+    const sortedWorkerPoolsConnection = this.createSortedWorkerPoolsConnection(
+      workerPoolsConnection,
       sortBy,
-      sortDirection,
-      searchTerm,
-      includeDeleted
+      sortDirection
     );
     const headers = [
-      { label: 'Worker Pool ID', id: 'workerPoolId', type: 'string' },
-      { label: 'Provider ID', id: 'providerId', type: 'string' },
-      {
-        label: 'Pending Tasks',
-        id: 'pendingTasks',
-        type: 'number',
-      },
-      {
-        label: 'Owner',
-        id: 'owner',
-        type: 'string',
-      },
-      {
-        label: '',
-        id: '',
-        type: 'undefined',
-      },
+      'Worker Pool ID',
+      'Provider ID',
+      'Pending Tasks',
+      'Owner',
+      '',
     ];
 
     return (
       <Fragment>
-        <DataTable
-          items={sortedWorkerPools}
+        <ConnectionDataTable
+          connection={sortedWorkerPoolsConnection}
+          pageSize={VIEW_WORKER_POOLS_PAGE_SIZE}
           headers={headers}
-          sortByLabel={sortBy}
+          sortByHeader={sortBy}
```

Comment:
It looks like the sorting is not working. The fix should probably be in `handleHeaderClick`.